File: Documentation/5.4/Samples/python/ClientApi/Operations/Maintenance/Indexes/GetIndexNames.py
from ravendb import GetIndexNamesOperation
from ravendb.documents.operations.definitions import MaintenanceOperation

from examples_base import ExampleBase


class GetIndexNames(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_get_index_names(self):
        with self.embedded_server.get_document_store("IndexNames") as store:
            # region get_index_names
            # Define the get index names operation
            # Pass number of indexes to skip & number of indexes to retrieve
            get_index_names_op = GetIndexNamesOperation(0, 10)

            # Execute the operation by passing it to maintenance.send
            index_names = store.maintenance.send(get_index_names_op)

            # index_names will contain the first 10 indexes, alphabetically ordered
            # endregion

    class Foo:
        # region get_index_names_syntax
        class GetIndexNamesOperation(MaintenanceOperation):
            def __init__(self, start: int, page_size: int): ...

        # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/Maintenance/Indexes/GetIndexNames.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Maintenance\Indexes;

use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Operations\Indexes\GetIndexNamesOperation;
use RavenDB\Type\StringArrayResult;

class GetIndexNames
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region get_index_names
            // Define the get index names operation
            // Pass number of indexes to skip & number of indexes to retrieve
            $getIndexNamesOp = new GetIndexNamesOperation(0, 10);

            // Execute the operation by passing it to Maintenance.Send
            /** @var StringArrayResult $indexNames */
            $indexNames = $store->maintenance()->send($getIndexNamesOp);

            // indexNames will contain the first 10 indexes, alphabetically ordered
            # endregion
        } finally {
            $store->close();
        }
    }

}
